File: backend_go/dto/task.go
package dto

// task dto to manage lifecycle of the request
// Redis bound

type ReviewTask struct {
	Type     string `json:"type"`
	UserID   uint   `json:"user_id"`
	ReviewID uint   `json:"review_id"`
	Code     string `json:"code"`
	Action   string `json:"action"`
}

type EnhanceTask struct {
	Type          string `json:"type"`
	UserID        uint   `json:"user_id"`
	EnhancementID uint   `json:"enhancement_id"`
	Code          string `json:"code"`
	Action        string `json:"action"`
}

// internal - worker input
type AssistantTask struct {
	Type           string `json:"type"`
	UserID         uint   `json:"user_id"`
	ConversationID uint   `json:"conversation_id"`
	Prompt         string `json:"prompt"`
}

// worker output
type AssistantTaskEvent struct {
	Type           string `json:"type"` //assistant.chunk | assistant.completed
	UserID         uint   `json:"user_id"`
	ConversationID uint   `json:"conversation_id"`
	Chunk          string `json:"chunk,omitempty"`
	Content        string `json:"content,omitempty"`
}

File: backend_python/schemas/dto/task.py
from pydantic import BaseModel
from typing import Union, Literal


class ReviewTask(BaseModel):
    type: Literal["review"] 
    user_id: int 
    review_id: int
    code: str
    action: str = "generate_summary"


class EnhancementTask(BaseModel):
    type: Literal["enhance"]
    user_id: int
    enhancement_id: int 
    code: str
    action: str = "enhance_code"


class ChatTask(BaseModel): 
    type: Literal["assistant"]
    user_id: int
    conversation_id: int
    prompt: str


Task = Union[ReviewTask, EnhancementTask, ChatTask]